token-minter/main.go

```diff
@@ -65,106 +65,113 @@ func main() {
 		os.Exit(1) // second signal. Exit directly.
 	}()
 
-	expirationTimestamp := mintToken(ctx, opts)
-
 	if opts.oneshot {
-		return
+		_, err := mintToken(ctx, opts)
+		if err != nil {
+			log.Fatalln(err)
+		}
+		os.Exit(0)
 	}
 
+	renewDuration := 1 * time.Second
 	for {
-		renewDuration := renewDuration(expirationTimestamp)
-		fmt.Println("renew delay set for", renewDuration.String())
 		select {
 		case <-time.After(renewDuration):
-			expirationTimestamp = mintToken(ctx, opts)
+			log.Println("minting token")
+			expirationTimestamp, err := mintToken(ctx, opts)
+			if err != nil {
+				log.Println("error minting token, will retry in", ErrorRetryPeriod, err)
+				renewDuration = ErrorRetryPeriod
+			} else {
+				log.Println("renew delay set for", renewDuration.String())
```

Comment:
Doesn't this log need to be after actually updating `renewDuration` which currently happens afterwards?

Reply:
Oops! Yes!

Reply:
Fixed